### ISSUE TITLE
cant connect to development server on android physical device for react-native init app

### ISSUE THREAD

**Issue by aravindle** [callstack/repack#415]
## Environment

OS: Windows
NPM version: 9.6.3
Node version: 19
React Native version: 0.72
Android Device: Realme c31

## Description

I created a basic react-native app with npx react-native init, and completed manual installation specified in the documentation. then started the server with npx react-native webpack-start and ran the application with npx react-native run-android and then getting the error "Cannot connect to the development error". Attaching the screenshot. I also tried adb reverse commands and changing the bundle location from the debug menu. Please help me fix this issue.

![WhatsApp Image 2023-08-19 at 17 09 37](https://github.com/callstack/repack/assets/139848214/85dc6087-4495-482f-8b3f-fe4e0dcb6b97)

## Reproducible Demo

https://github.com/aravindle/callstack-rn

**Comment by masonsi-dev** [comment]
I used the auto-setup and I am having same issue. Press cmd + m to open the dev console on android. And change the host to [::1]:8081 will help you connect to the dev server. But even successfully connect to the server, I encountered another weird error 
![CleanShot 2023-08-21 at 14 24 54@2x](https://github.com/callstack/repack/assets/101320274/815f181d-2b2e-4b60-a1f6-170e08b3ef30)

**Comment by aravindle** [comment]
Hi thankyou for suggestion, the error got fixed but a new error is starting to show. 
`Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'`

I checked for it and only suggestion was to upgrade node version. so i created a new app with node 20.5.1 version and the error is still persisting. anyone  has suggestions? thankyou..


![Screenshot 2023-08-27 082344](https://github.com/callstack/repack/assets/139848214/9081f8fa-ebaf-44a5-8386-6b468ee81bb3)
